### PR TITLE
feat(cardDescriptionPage)created UI and done functionality

### PR DESCRIPTION
**Pull request by Joelzac537**
**Introduction**
Created card description
**Description**
Created card description UI and then created functionality of the card description page.
**How To Test**
Created UI in index.html . Then created functionality in frames.js, created the API for card description in index.js .

### CODE REVIEW

**Review by viv9k — CHANGES_REQUESTED** [review]
Please fix all the required changes and update asap

File: functions/index.js [hunk 1]
```diff
@@ -193,4 +194,50 @@ function createSprintId(createNewTaskSprintNumber) {
...
+//     cors(request, response, () => {
+//         console.log(request);
+//         var title = request.body.data.Title;
+//         var des = request.body.data.Description;
+//         var priority = request.body.data.Priority;
+//         var difficulty = request.body.data.Difficulty;
+//         var creator = request.body.data.Creator;
+//         var assignee = request.body.data.Assignee;
+//         var estimatedTime = request.body.data.EstimatedTime;
+//         var status = request.body.data.Status;
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply functions/index.js:209]: removed

File: public/index.html [hunk 2]
```diff
@@ -236,7 +236,7 @@ <h6 class="card-text col-md-10 col-8" id="sprintStatus" style="font-size: 12px;"
         <div class="container pt-2 pb-5 pl-2">
             <h3 class="ml-1 fontHeading">Tasks</h3>
             <hr class="w-75 ml-0">
-            <div id="cardsList">
+            <div id="cardsList" type="button">
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/index.html:239]: removed

File: public/index.html [hunk 3]
```diff
@@ -444,6 +444,207 @@ <h3 class="pt-5 fontHeading text-white">Log Work</h3>
...
+            <div class="row mt-5">
+                <div class="col-md-11 col-10"></div>
+                <span type="button" class="col-md-1 col-2 mt-5 material-icons" id="backToMainFromCardDescription">
+                       close
+                    </span>
+            </div>
+            <div class="row">
+                <div class="alert form-inline w-50 col-md-4 col-5 mb-5" style="font-size: 20px;" role="alert">
+                    <div style="font-weight: bold;" class="mr-3">Date:</div>
+                    <div id="creationDateCreateNewTask"></div>
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/index.html:459]: changed

File: public/index.html [hunk 4]
```diff
@@ -444,6 +444,207 @@ <h3 class="pt-5 fontHeading text-white">Log Work</h3>
...
+                    <button type="submit" class="btn btn-primary mt-2 mb-5 mx-3" id="#">Log Work</button>
+                </div>
+                <div class="col-md-1 col-2">
+                    <button type="submit" class="btn btn-primary mt-2 mb-5" id="editPage">Edit</button>
+                </div>
+            </div>
+            <div class="form-group row mt-2">
+                <label for="input" class="col-sm-2 col-form-label">Title</label>
+                <div class="alert alert-info form-inline col-sm-10" role="alert">
+                    <div id="framesDetailsTitle"></div>
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/index.html:472]: changed

File: public/index.html [hunk 5]
```diff
@@ -444,6 +444,207 @@ <h3 class="pt-5 fontHeading text-white">Log Work</h3>
...
+                </div>
+            </div>
+            <div class="form-group row">
+                <label for="description" class="col-sm-2 col-form-label">Description</label>
+                <div class="alert alert-info form-inline col-sm-10" role="alert">
+                    <div id="framesDetailsDescription"></div>
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/index.html:478]: changed

File: public/index.html [hunk 6]
```diff
@@ -444,6 +444,207 @@ <h3 class="pt-5 fontHeading text-white">Log Work</h3>
...
+                </div>
+            </div>
+            <div class="form-row">
+                <div class="form-group col-md-4">
+                    <label for="input">Assignee</label>
+                    <div class="alert alert-info form-inline" role="alert">
+                        <div id="framesDetailsAssignee"></div>
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/index.html:485]: changed

File: public/index.html [hunk 7]
```diff
@@ -444,6 +444,207 @@ <h3 class="pt-5 fontHeading text-white">Log Work</h3>
...
+                <div class="form-group col-md-4">
+                    <label for="text">Creator</label>
+                    <div class="alert alert-info form-inline" role="alert">
+                        <div id="framesDetailsCreator"></div>
+                    </div>
+                </div>
+                <div class="form-group col-md-4">
+                    <label for="text">Estimated Time</label>
+                    <div class="alert alert-info form-inline" role="alert">
+                        <div id="framesDetailsEstimatedTime"></div>
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/index.html:497]: changed

File: public/index.html [hunk 8]
```diff
@@ -444,6 +444,207 @@ <h3 class="pt-5 fontHeading text-white">Log Work</h3>
...
+                <div class="form-group col-md-4">
+                    <label for="input">Assignee</label>
+                    <div class="alert alert-info form-inline" role="alert">
+                        <div id="framesDetailsAssignee"></div>
+                    </div>
+                </div>
+                <div class="form-group col-md-4">
+                    <label for="text">Creator</label>
+                    <div class="alert alert-info form-inline" role="alert">
+                        <div id="framesDetailsCreator"></div>
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/index.html:491]: changed

File: public/index.html [hunk 9]
```diff
@@ -444,6 +444,207 @@ <h3 class="pt-5 fontHeading text-white">Log Work</h3>
...
+                    <div class="alert alert-info form-inline" role="alert">
+                        <div id="framesDetailsEstimatedTime"></div>
+                    </div>
+                </div>
+            </div>
+            <div class="form-row">
+                <div class="form-group col-md-4">
+                    <label for="input">Category</label>
+                    <div class="alert alert-info form-inline" role="alert">
+                        <div id="framesDetailsCategory"></div>
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/index.html:505]: changed

File: public/index.html [hunk 10]
```diff
@@ -444,6 +444,207 @@ <h3 class="pt-5 fontHeading text-white">Log Work</h3>
...
+                    </div>
+                </div>
+                <div class="form-group col-md-4">
+                    <label for="text">Priority</label>
+                    <div class="alert alert-info form-inline" role="alert">
+                        <div id="framesDetailsPriority"></div>
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/index.html:511]: changed

File: public/index.html [hunk 11]
```diff
@@ -444,6 +444,207 @@ <h3 class="pt-5 fontHeading text-white">Log Work</h3>
...
+                    </div>
+                </div>
+                <div class="form-group col-md-4">
+                    <label for="text">Difficulty</label>
+                    <div class="alert alert-info form-inline" role="alert">
+                        <div id="framesDetailsDifficulty"></div>
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/index.html:517]: changed

File: public/index.html [hunk 12]
```diff
@@ -444,6 +444,207 @@ <h3 class="pt-5 fontHeading text-white">Log Work</h3>
...
+                    </div>
+                </div>
+            </div>
+            <div class="form-row">
+                <div class="form-group col-md-4">
+                    <label for="inputState">Status</label>
+                    <div class="alert alert-info form-inline" role="alert">
+                        <div id="framesDetailsStatus"></div>
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/index.html:525]: changed

File: public/index.html [hunk 13]
```diff
@@ -444,6 +444,207 @@ <h3 class="pt-5 fontHeading text-white">Log Work</h3>
...
+                    </div>
+                </div>
+                <div class="form-group col-md-4">
+                    <label for="inputState">Logged Hours</label>
+                    <div class="alert alert-info form-inline" role="alert">
+                        <div id="framesDetailsLogHours"></div>
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/index.html:531]: changed

File: public/index.html [hunk 14]
```diff
@@ -444,6 +444,207 @@ <h3 class="pt-5 fontHeading text-white">Log Work</h3>
...
+                    </div>
+                </div>
+                <div class="form-group col-md-4">
+                    <label for="inputState">Work Done</label>
+                    <div class="alert alert-info form-inline" role="alert">
+                        <div id="framesDetailsWorkDone"></div>
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/index.html:537]: changed

File: public/index.html [hunk 15]
```diff
@@ -444,6 +444,207 @@ <h3 class="pt-5 fontHeading text-white">Log Work</h3>
...
+                    </div>
+                </div> -->
+            </div>
+        </div>
+    </div>
+
+    <!-- edit Page-------------------------------------------------------- -->
+    <div class="container py-5" id="editTaskPage">
+        <div class="bg-light px-5">
+            <div class="row pt-2">
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/index.html:555]: removed

File: public/scripts/frames.js [hunk 16]
```diff
@@ -59,29 +59,43 @@ function removeInstance(data) {
     });
 }
 
+
+
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/scripts/frames.js:63]: removed

File: public/scripts/frames.js [hunk 17]
```diff
@@ -59,29 +59,43 @@ function removeInstance(data) {
...
         var priority = dataObj.priority;
         var assignee = dataObj.assignee;
         var creator = dataObj.creator;
         var category = dataObj.category;
         var status = dataObj.status;
+        var logWorkTotalTime = dataObj.logWorkTotalTime;
+        var workDone = dataObj.workDone;
+        var creationDate = dataObj.creationDate;
+        var createNewTaskSprintNumber = dataObj.createNewTaskSprintNumber;
+        var storyPointerNumber = dataObj.storyPointerNumber;
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/scripts/frames.js:85]: checked and corrected changes

File: public/scripts/worktrollyUI.js [hunk 18]
```diff
@@ -88,6 +88,29 @@ $("#logWork").ready(function() {
...
+});
+
+$("#editPage").click(function() {
+    newPage = "editTaskPage";
+    uiLoader();
+});
+
+$("#editTaskPage").ready(function() {
+    $("#editTaskPage").hide(0);
+});
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/scripts/worktrollyUI.js:107]: removed

File: public/scripts/worktrollyUI.js [hunk 19]
```diff
@@ -136,4 +159,37 @@ $("#submitNewSprint").click(function() {
...
+    var des = $("#editDetailsDescription").val();
+    var priority = $("#editDetailsPriority").val();
+    var difficulty = $("#editDetailsDifficulty").val();
+    var creator = $("#editDetailsCreator").val();
+    var assignee = $("#editDetailsAssignee").val();
+    var estimatedTime = $("#editDetailsEstimatedTime").val();
+    var status = $("#editDetailsStatus").val();
+    var category = $("#editDetailsCategory").val();
+    // var storyPointNumber = $("#editDetailsStoryPoint").val();
+    // var sprintNumber = $("#editDetailsSprintNumber").val();
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/scripts/worktrollyUI.js:175]: removed

File: public/scripts/frames.js [hunk 20]
```diff
@@ -128,4 +142,79 @@ function setDataIntoCard() {
...
+    document.getElementById("framesDetailsDescription").innerHTML = des;
+    document.getElementById("framesDetailsPriority").innerHTML = priority;
+    document.getElementById("framesDetailsAssignee").innerHTML = assignee;
+    document.getElementById("framesDetailsCreator").innerHTML = creator;
+    document.getElementById("framesDetailsCategory").innerHTML = category;
+    document.getElementById("framesDetailsStatus").innerHTML = status;
+    document.getElementById("framesDetailsEstimatedTime").innerHTML = estimatedTime;
+    document.getElementById("framesDetailsDifficulty").innerHTML = difficulty;
+    document.getElementById("framesDetailsLogHours").innerHTML = logHours;
+    document.getElementById("framesDetailsWorkDone").innerHTML = workDone;
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/scripts/frames.js:170]: updated

File: public/scripts/frames.js [hunk 21]
```diff
@@ -128,4 +142,79 @@ function setDataIntoCard() {
...
+
+// function setEditDetail(Title) {
+//     dataset.forEach(dataObj => {
+//         var title = dataObj.title;
+//         var des = dataObj.des;
+//         var priority = dataObj.priority;
+//         var assignee = dataObj.assignee;
+//         var creator = dataObj.creator;
+//         var category = dataObj.category;
+//         var status = dataObj.status;
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/scripts/frames.js:184]: removed

File: public/scripts/frames.js [hunk 22]
```diff
@@ -59,29 +59,43 @@ function removeInstance(data) {
...
+        var logHours = dataObj.logHours;
         var priority = dataObj.priority;
         var assignee = dataObj.assignee;
         var creator = dataObj.creator;
         var category = dataObj.category;
         var status = dataObj.status;
+        var logWorkTotalTime = dataObj.logWorkTotalTime;
+        var workDone = dataObj.workDone;
+        var creationDate = dataObj.creationDate;
+        var createNewTaskSprintNumber = dataObj.createNewTaskSprintNumber;
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/scripts/frames.js:84]: updated

File: public/scripts/worktrollyClassData.js [hunk 23]
```diff
@@ -15,4 +16,4 @@ class StoreData {
         this.createNewTaskSprintNumber = data.CreateNewTaskSprintNumber;
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/scripts/worktrollyClassData.js:16]: updated

File: public/scripts/frames.js [hunk 24]
```diff
@@ -59,29 +59,43 @@ function removeInstance(data) {
...
+        var storyPointerNumber = dataObj.storyPointerNumber;
         var tickets = document.getElementById("cardsList").innerHTML;
 
         if (category == selectedCategory) {
 
-            var frame = "<div class=\"card text-white bg-dark mb-4\" style=\"max-width: 32rem;\">";
+
+            var frame = "<div class=\"card text-white bg-dark mb-4\" onclick=\"showDescription('" + id + "','" + title + "','" + des + "','" + estimatedTime + "','" + difficulty + "','" + logHours + "','" + priority + "','" + assignee + "','" + creator + "','" + category + "','" + status + "','" + logWorkTotalTime + "','" + workDone + "','" + creationDate + "','" + createNewTaskSprintNumber + "','" + storyPointerNumber + "')\" style=\"max-width: 32rem;\">";
```
[the remark this answers is not in the document]
Comment (Joelzac537) [reply public/scripts/frames.js:91]: updated

File: public/index.html [hunk 25]
```diff
@@ -372,7 +372,7 @@ <h3 class="pl-1 fontHeading">Create New Task</h3>
                 </div>
                 <div class="form-group col-md-4">
                     <label for="inputState">Sprint Number</label>
-                    <input type="number" class="form-control" id="createNewTaskSprintNumber">
+                    <input type="number" class="form-control" id="sprintNumber">
```
Comment (viv9k) [inline public/index.html:375]: Here you need not to change the id because this belongs to create new task

File: public/index.html [hunk 26]
```diff
@@ -444,6 +444,110 @@ <h3 class="pt-5 fontHeading text-white">Log Work</h3>
...
+                    </div>
+                </div>
+                <!-- <div class="form-row">
+                    <div class="form-group col-md-4">
+                        <label for="inputState">Creation Date</label>
+                        <div class="alert alert-info form-inline" role="alert">
+                            <div id="framesDetailsCreationDate"></div>
+                        </div>
+                    </div>
+                </div> -->
```
Comment (viv9k) [inline public/index.html:547]: Remove commented code

**Review by viv9k — CHANGES_REQUESTED** [review]
Please fix these things and your code is ready to merge

File: public/index.html (hunk 25 above)
Comment (Joelzac537) [reply public/index.html:375]: updated

File: public/index.html (hunk 26 above)
Comment (Joelzac537) [reply public/index.html:547]: removed

**Review by viv9k — APPROVED** [review]

**Pull request merged**